### PR TITLE
Redis data source implementation

### PR DESCRIPTION
**Pull request by sczyh30**
Add Redis data source service type implementation as well as document.

### CODE REVIEW

File: vertx-service-discovery/src/main/java/io/vertx/servicediscovery/types/RedisDataSource.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * @author Eric Zhao
+ */
+@VertxGen
+public interface RedisDataSource extends DataSource {
+
+  String TYPE = "datasource.redis";
+
+  String DEFAULT_TYPE = "redis";
+
+  static Record createRecord(String name, JsonObject location, JsonObject metadata) {
```
Comment (cescoffier) [inline vertx-service-discovery/src/main/java/io/vertx/servicediscovery/types/RedisDataSource.java]: can you add javadoc ?

**Comment by cescoffier** [comment]
Looks good except this javadoc issue.

Did you sign the Eclipse CLA ? (https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md)

**Comment by sczyh30** [comment]
Yes, I've added javadoc, and signed the Eclipse CLA :-)

**Comment by cescoffier** [comment]
Perfect, thanks !

**Pull request merged**